### PR TITLE
Add comments explaining several types of domains.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Review by zasdfgbnm — APPROVED** [review]

File: csrc/ir/internal_base_nodes.h [hunk 1]
```diff
@@ -554,18 +554,26 @@ class TensorDomain : public Val {
     return no_bcast_domain_;
   }
 
+  // The input logical domain. The root domain of a consumer should equal the
+  // rfactor domain of its producer.
```
Comment (jacobhinkle) [inline csrc/ir/internal_base_nodes.h:558]: "... ignoring IterDomains of type Reduction."

File: csrc/ir/internal_base_nodes.h [hunk 2]
```diff
@@ -554,18 +554,26 @@ class TensorDomain : public Val {
...
     return rfactor_domain_;
   };
 
+  // The allocation domain. This describes how data is stored in memory in
+  // outer-to-inner order.
   const std::vector<IterDomain*>& allocation() const {
     return allocation_domain_;
   }
 
+  // The index domain after scheduling. nvFuser generates loop indices according
```
Comment (naoyam) [inline csrc/ir/internal_base_nodes.h:575]: "Index domain" may be confusing. Leaf domains just define loop nests and loop indices, but final index expressions are generally generated for allocation domains.

Comment (wujingyue) [reply csrc/ir/internal_base_nodes.h]: Fixed.

File: csrc/ir/internal_base_nodes.h (hunk 1 above)
Comment (wujingyue) [reply csrc/ir/internal_base_nodes.h]: Fixed.

**Review by naoyam — APPROVED** [review]
LGTM. Thanks for adding the comments.

**Pull request merged**